community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/TransactionStateMachine.java

```diff
@@ -386,6 +422,8 @@ private static BoltResultHandle executeQuery( MutableTransactionState ctx, SPI s
         /** The current transaction, if present */
         KernelTransaction currentTransaction;
 
+        Optional<Status> pendingTerminationNotice = Optional.empty();
```

Comment:
Maybe it's better to have this field without `Optional`? It is only used using `#isPresent()` + `#get()` which is essentially a null-check.

Reply:
You're right.